pkg/storage/tsdb/bucketindex/index.go

```diff
@@ -0,0 +1,195 @@
+package bucketindex
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/oklog/ulid"
+	"github.com/prometheus/prometheus/tsdb"
+	"github.com/thanos-io/thanos/pkg/block"
+	"github.com/thanos-io/thanos/pkg/block/metadata"
+
+	cortex_tsdb "github.com/cortexproject/cortex/pkg/storage/tsdb"
+	"github.com/cortexproject/cortex/pkg/util"
+)
+
+const (
+	IndexFilename = "bucket-index.json"
+	IndexVersion1 = 1
+
+	SegmentsFormatUnknown       = ""
+	SegmentsFormat1Based6Digits = "1b6d"
```

Comment:
Can we add a comment on what this format is? This looks quite cryptic without explanation.

Reply:
Sure